cpp/lib/src/Elasticsearch.cc

```diff
@@ -0,0 +1,157 @@
+/** \file    Elasticsearch.cc
+ *  \brief   Implementation of utility functions relating to Elasticsearch.
+ *  \author  Mario Trojan
+ *
+ *  \copyright 2018 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "Elasticsearch.h"
+#include <cstdlib>
+
+
+std::shared_ptr<JSON::ObjectNode> Elasticsearch::FieldsToJSON(const Fields fields) {
+    std::shared_ptr<JSON::ObjectNode> tree_root(new JSON::ObjectNode);
+    for (const auto &field : fields) {
+        JSON::StringNode * value_node = new JSON::StringNode(field.second);
```

Comment:
No space after the asterisk and variable initialisation with parentheses please.